build/build-jemalloc-binaries.sh

```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+# Build cockroach binary using jemalloc as the allocator.
+
+set -euo pipefail
+
+source $(dirname $0)/build-common.sh
+
+# This is mildly tricky: This script runs itself recursively. The
```

Comment:
Now that we have three of these it's time to stop duplicating this "tricky" code. How about removing everything related to docker from the `build-*-binaries` scripts and call them from circle.yml as `build/builder.sh build/build-jemalloc-binaries.sh`?